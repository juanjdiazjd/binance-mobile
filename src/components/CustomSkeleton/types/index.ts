export interface FigureStyle {
  type: FigureType;
  size: FigureSize;
  width: number;
  height: number;
  borderRadius?: number;
}

export enum FigureSize {
  large = 'big',
  medium = 'medium',
  small = 'small',
}

export enum SkeletonType {
    text = 'text',
    image = 'image',
  }
export enum FigureType {
  circle = 'circle',
  rectangle = 'rectangle',
}
export const figureValues: FigureStyle[] = [
  {
    type: FigureType.circle,
    size: FigureSize.small,
    width: 30,
    height: 30,
    borderRadius: 100,
  },
  {
    type: FigureType.circle,
    size: FigureSize.medium,
    width: 60,
    height: 60,
    borderRadius: 100,
  },
  {
    type: FigureType.circle,
    size: FigureSize.large,
    width: 80,
    height: 80,
    borderRadius: 100,
  },
  {
    type: FigureType.rectangle,
    size: FigureSize.small,
    width: 30,
    height: 10,
  },
  {
    type: FigureType.rectangle,
    size: FigureSize.medium,
    width: 60,
    height: 10,
  },
  {
    type: FigureType.rectangle,
    size: FigureSize.large,
    width: 80,
    height: 10,
  },
];
