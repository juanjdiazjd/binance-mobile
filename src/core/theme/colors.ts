const colors = {
  primary: '#F2BC07',
  secondary: '#FAD338',
  primaryDark: '#F0B10F',
  titleText: '#FFFFFF',
  titleTextSecondary: '#8b9dc3',
  gray: '#1E2329',
  charcoalGray: '#36454F',
  black: '#0B0E11',
  white: '#FFFFFF',
  green: '#27ae60',
  red: '#ff0050',
  iris: '#3D52D5',
  background: '#1E2329',
  transparent: 'transparent',
  blackTransparent: 'rgba(52, 52, 52, 0.2)',
};
export default colors;
