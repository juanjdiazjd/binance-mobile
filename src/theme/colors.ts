const colors = {
primary: '#FED428',
secondary: '#21232B',
primaryDark:'#ffe473',
darkGray: '#3D434A',
gray:'#D8D8D8',
black:'#000000',
white: '#FFFFFF',
green: '#27ae60',
red: '#e74c3c',
iris: '#3D52D5',
background: '#003366',
transparent: 'transparent',
blackTransparent: 'rgba(52, 52, 52, 0.2)'
}
export default colors