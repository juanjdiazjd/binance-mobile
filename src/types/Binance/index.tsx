export type ResponseTickerPrice = {
  symbol: string;
  price: string;
};
