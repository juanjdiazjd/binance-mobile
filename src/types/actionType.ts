
export type actionType<T = {}> = {
    type: string;
    error?: boolean;
    payload?: T;
  };

  
export const createAction = ({ type, error, payload }: actionType<any>) => {
    return {
      type,
      error,
      payload
    };
  };
  