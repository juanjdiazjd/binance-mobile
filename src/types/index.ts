export * from './Home';
export * from './Navigation';
export * from './Sizes';
export * from './ListItem';
export * from './ListCryptoItem';
export * from './Exchange';
export * from './Order';
