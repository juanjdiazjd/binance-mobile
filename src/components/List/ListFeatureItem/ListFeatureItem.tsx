import * as React from 'react';
import {Image, Text, TouchableOpacity, View, ViewProps} from 'react-native';
import styled from 'styled-components';

import Icon from 'react-native-vector-icons/MaterialIcons';
import {Feature} from './types/ListFeature';
import theme from 'core/theme';

const ComponentContainer = styled(TouchableOpacity)`
  flex-direction: row;
  justify-content: center;
  margin-bottom: 10px;
  height: 115px;
  width: auto;
`;

const ListContainer = styled(View)`
  background-color: ${theme.colors.charcoalGray};
  width: 320px;
  flex-direction: row;
  border-radius: 20px;
  shadow-opacity: 0.3;
  shadow-offset: 0px 4px;
  shadow-radius: 4px;
  elevation: 3;
  justify-content: space-between;
`;

const ColorBar = styled(View)<{
  color: string;
}>`
  width: 0px;
  border-top-start-radius: 20px;
  border-bottom-start-radius: 20px;
  padding: 4%;
  background-color: ${({color}) => color ?? theme.colors.primary};
`;

export const TextTitle = styled(Text)<{
  color?: string;
}>`
  color: ${({color}) => color ?? theme.colors.titleText};
  white-space: pre-line;
  max-width: 120px;
  font-size: 20px;
  align-self: center;
  font-weight: bold;
`;

const AvatarContainer = styled(View)<{
  backgroundColor: string;
}>`
  width: 90px;
  height: 90px;
  color: ${({backgroundColor}) => backgroundColor ?? theme.colors.background};
  border-radius: 80px;
  align-self: center;
`;

const Avatar = styled(Image)`
  width: 70px;
  height: 70px;
  top: 10%;
  resize-mode: contain;
  align-self: center;
`;
const CustomIcon = styled(Icon)`
  align-self: center;
  padding: 10px;
`;

interface ListFeatureItemViewProps extends ViewProps {
  item: Feature;
  onHandlePress: () => void;
}

export const ListFeatureItem: React.FunctionComponent<
  ListFeatureItemViewProps
> = ({item, onHandlePress}) => {
  return (
    <ComponentContainer onPress={onHandlePress}>
      {item && (
        <ListContainer>
          <ColorBar color={item.color} />
          <AvatarContainer backgroundColor={item.backgroundColor}>
            <Avatar
              source={{
                uri: `${item.image}`,
              }}
            />
          </AvatarContainer>
          <TextTitle color={theme.colors.white}>{item.title}</TextTitle>
          <CustomIcon
            name="arrow-forward-ios"
            size={30}
            color={theme.colors.white}
          />
        </ListContainer>
      )}
    </ComponentContainer>
  );
};

export default ListFeatureItem;
