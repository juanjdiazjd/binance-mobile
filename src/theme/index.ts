
import baseStyled, { ThemedStyledInterface } from 'styled-components';
import colors from './colors';

const theme = {
  colors: {
    ...colors
  },
  sizes: {
    contentWidth: '90%',
    fullWidth: '100%',
    fullHeight: '100%'
  },
  fonts: {
    default: 'Montserrat-Light',
    robotoLight: 'Roboto-Light',
    robotoBold: 'Roboto-Bold',
    robotoMedium: 'Roboto-Medium',
    montserratSemiBold: 'Montserrat-SemiBold'
  }
};

export default theme;

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
