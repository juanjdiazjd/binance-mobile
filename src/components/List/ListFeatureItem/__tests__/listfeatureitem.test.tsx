import theme from 'core/theme';
import React from 'react';
import renderer from 'react-test-renderer';
import {Feature, ItemsHomeStackList} from 'types';
import ListFeatureItem from '../ListFeatureItem';

jest.mock('swr');
jest.mock('react-native-localization', () => {
  return {
    getLocales: jest.fn(),
  };
});

jest.resetAllMocks();
jest.useFakeTimers();

const MOCK_DATA_LIST_FEATURE_ITEM: Feature = {
  id: 1,
  image: 'https://i.imgur.com/a6qfBLb.png',
  title: 'Exchange',
  screenName: ItemsHomeStackList.Exchange,
  color: theme.colors.primary,
  backgroundColor: theme.colors.primaryDark,
};

const ListFeatureItemComponent = (): JSX.Element => (
  <ListFeatureItem
    item={MOCK_DATA_LIST_FEATURE_ITEM}
    onHandlePress={jest.fn()}
  />
);

it('Should list feature item component', () => {
  const tree = renderer.create(ListFeatureItemComponent()).toJSON();
  expect(tree).toMatchSnapshot();
});
