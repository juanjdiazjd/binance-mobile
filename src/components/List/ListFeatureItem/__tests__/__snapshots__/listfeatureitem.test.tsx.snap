// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should list feature item component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    className="sc-gsnTZi dnvkNT"
  >
    <View
      className="sc-dkzDqf jIYPSQ"
      color="#F2BC07"
    />
    <View
      backgroundColor="#F0B10F"
      className="sc-eCYdqJ cNakAB"
    >
      <Image
        className="sc-jSMfEi kIrGFD"
        source={
          Object {
            "uri": "https://i.imgur.com/a6qfBLb.png",
          }
        }
      />
    </View>
    <Text
      className="sc-hKMtZM iSrUnU"
      color="#FFFFFF"
    >
      Exchange
    </Text>
    <Text
      allowFontScaling={false}
      className="sc-gKXOVf WzVOL"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î—¡
    </Text>
  </View>
</View>
`;
